src/main.ts

```diff
@@ -194,6 +199,42 @@ export function createEmitter<T extends Events>(options?: EmitterOptions) {
     }))
   }
 
+  function addGlobalHandler(globalEventHandler: GlobalEventHandler<T>, options?: EmitterOnOptions): () => void {
+    const { signal } = options ?? {}
+
+    if(signal?.aborted) {
+      return () => {}
+    }
+
+    globalHandlers.add(globalEventHandler)
+
+    signal?.addEventListener('abort', () => off(globalEventHandler))
+
+    return () => off(globalEventHandler)
+  }
+
+  function addEventHandler<E extends Event>(event: E, handler: Handler<EventPayload<E>>, options?: EmitterOnOptions): () => void {
+    const { signal } = options ?? {}
+
+    if(signal?.aborted) {
+      return () => {}
+    }
+
+    const existing = handlers.get(event)
+
+    if (existing) {
+      existing.add(handler)
+    } else {
+      handlers.set(event, new Set([handler]))
+    }
+
+    if(signal) {
+      signal.addEventListener('abort', () => off(event, handler))
+    }
+
+    return () => off(event, handler)
```

Comment:
if you liked the suggestion above

```suggestion
    const offHandler = () => off(event, handler)

    signal?.addEventListener('abort', offHandler)

    return offHandler
```

Reply:
if you didn't, I noticed the global version uses `signal?.addEventListener` and this one has a separate line for `if(signal) { ... }`. Not sure if that's intentional

Reply:
Wasn't intentional 👍